EdFi.Ods.Utilities.Migration.Tests/Utilities/DatabaseDeploymentProvider.cs

```diff
@@ -0,0 +1,108 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using EdFi.Ods.Utilities.Migration.Tests.MigrationTests;
+using log4net;
+
+namespace EdFi.Ods.Utilities.Migration.Tests.Utilities
+{
+    public class DatabaseDeploymentProvider : IDatabaseDeploymentProvider
+    {
+        private readonly ILog _logger = LogManager.GetLogger(typeof(DatabaseDeploymentProvider));
+        private readonly string _dbDeployName = MigrationTestSettingsProvider.GetConfigVariable("DbDeployName");
+        private readonly string _dbDeployVersion = MigrationTestSettingsProvider.GetConfigVariable("DbDeployVersion");
+        private readonly string _dbDeployPath = MigrationTestSettingsProvider.GetConfigVariable("DbDeployPath");
+        private readonly string _dbDeploySource = MigrationTestSettingsProvider.GetConfigVariable("DbDeploySource");
+
+        protected void LogErrorMessage(string msg) => _logger.Error(msg);
+
+        public void Deploy(string version, string verb, string engine, string database, string connectionString)
+        {
+            using (var ps = PowerShell.Create())

```

Comment:
should we be using powershell here? Why not just a process instead? That way we can be compliant across platforms?

Reply:
Ticket Created.